blobstore/wants.go

```diff
@@ -136,14 +165,29 @@ type wantProc struct {
 }
 
 func (proc *wantProc) init() {
+
+	proc.wmgr.promGauge("proc", 1)
+
 	bsCancel := proc.bs.Changes().Register(
 		luigi.FuncSink(func(ctx context.Context, v interface{}, err error) error {
 			proc.l.Lock()
 			defer proc.l.Unlock()
+			if luigi.IsEOS(err) {
+				return nil
+			}
+
+			if v == nil {
+				proc.wmgr.info.Log("event", "wantProc notification", "warn", "nil value", "goterr", err)
+				return nil
+			}
 
-			notif := v.(ssb.BlobStoreNotification)
+			notif, ok := v.(ssb.BlobStoreNotification)
+			if !ok {
+				proc.wmgr.info.Log("event", "wantProc notification", "err", "wrong type", "t", fmt.Sprintf("%T", v), "goterr", err)
+				return nil
```

Comment:
I feared this might stop processing?

I added this when I found the createWants leak. Once I closed the sink these funcs were called with `v==nil` and `err==luigi.EOS` which I now also check so this could be reverted back to the original or return an error.

Reply:
Good point! You're probably right, let's keep it like this.